### ISSUE TITLE
[4.8.1]: `{{ actionUrl() }}` Twig function changed behavior

### ISSUE THREAD

**Issue by joshuapease** [craftcms/cms#14862]
### Description

It appears that the behavior of `actionUrl` change in the Craft 4.8.1 update. It no longer forces `https` when no scheme is specified ([UrlHelper](https://github.com/craftcms/cms/blob/4.x/src/helpers/UrlHelper.php#L377)). But also... intentionally setting a scheme does nothing as well. 

I noticed this because I had some JS creating a `new URL()` object, and it started throwing errors since the URL was no longer absolute.

**Before: Craft 4.8.0**
```
{{ actionUrl('commerce/payments/complete-payment') }}

https://my-project.ddev.site/actions/commerce/payments/complete-payment

{{ actionUrl('commerce/payments/complete-payment', scheme='https') }}

https://my-project.site/actions/commerce/payments/complete-payment
```

**After: Craft 4.8.1**

```
{{ actionUrl('commerce/payments/complete-payment') }}

/actions/commerce/payments/complete-payment

## Forcing https doesn't work either.

{{ actionUrl('commerce/payments/complete-payment', scheme='https') }}

/actions/commerce/payments/complete-payment
```

**Worth noting**
Forcing a scheme on the `url()` function works as expected.

### Steps to reproduce

1. Try `actionUrl()` in Craft 4.8.0 and then 4.8.1

### Expected behavior

`actionUrl` will default to `https` when the conditions in UrlHelper are met (Stick with SSL if the current request is over SSL and a scheme wasn't defined). 

### Actual behavior

Automatic `https` and overriding `scheme` doesn't seem to work

### Craft CMS version

4.8.1

### PHP version

8.1

### Installed plugins and versions

-

**Comment by brandonkelly** [comment]
Craft 4.8.1 fixed a bug in `actionUrl()` where the front-end site URL was being used rather than the requested URL, which could be problematic on headless sites where Craft isn’t actually serving the front-end site. (#14440)

There were a couple follow-up fixes so if you’re still running 4.8.1, you should update to the latest 4.x release (currently 4.8.10).

If you’re still having issues after that, then the culprit is most likely that you are manually setting the `@web` alias from `config/general.php`, and it’s not getting set to the exact hostname being requested.

We are moving away from recommending that you set `@web` yourself, so if you’re in fact doing that, the easiest fix will be to stop. And make sure that your site’s base URL doesn’t start with `@web` — it should be getting explicitly set to a base URL, either directly or via an environment variable.

**Comment by joshuapease** [comment]
Ah! Yep! `@web` was the culprit. It was set to `/`
